idmapping-rest/src/test/java/org/uniprot/api/idmapping/repository/IdMappingRepositoryTest.java

```diff
@@ -0,0 +1,123 @@
+        storeManager.close();
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = SolrCollection.class,
+            names = {"uniprot", "uniparc", "uniref"})
+    void canSendQueryForInterestedCollections(SolrCollection collection)
+            throws IOException, SolrServerException {
+        IdMappingRepository idMappingRepository =
+                new IdMappingRepository(
+                        new MockSearchableSolrClient(), new MockSearchableSolrClient());
+        assertTrue(idMappingRepository.getAllMappingIds(collection, new ArrayList<>()).isEmpty());
+    }
+
+    @Test
+    void canGetUniParcMapping() throws SolrServerException, IOException {
+        var id1 = "UPI0000000001";
+        var id50 = "UPI0000000050";
+        var id91 = "UPI0000000091";
+
+        var mappedIdsPairs =
+                idMappingRepository.getAllMappingIds(
+                        SolrCollection.uniparc, List.of(id1, id50, id91));
+
+        assertAll(
+                () -> assertEquals(3, mappedIdsPairs.size()),
+                () -> assertEquals(id1, mappedIdsPairs.get(0).getFrom()),
+                () -> assertEquals(id1, mappedIdsPairs.get(0).getTo()),
+                () -> assertEquals(id50, mappedIdsPairs.get(1).getFrom()),
+                () -> assertEquals(id50, mappedIdsPairs.get(1).getTo()),
+                () -> assertEquals(id91, mappedIdsPairs.get(2).getFrom()),
+                () -> assertEquals(id91, mappedIdsPairs.get(2).getTo()));
+    }
+
+    @Test
+    void canGetUniRefMapping() throws SolrServerException, IOException {
+        var id100 = "UniRef100_UPI0000000100";
+        var id99 = "UniRef100_UPI0000000099";
+        var id98 = "UniRef100_UPI0000000098";
+
+        var mappedIdsPairs =
+                idMappingRepository.getAllMappingIds(
+                        SolrCollection.uniref, List.of(id100, id99, id98));
+
+        assertAll(
+                () -> assertEquals(3, mappedIdsPairs.size()),
+                () -> assertEquals(id100, mappedIdsPairs.get(2).getFrom()),
+                () -> assertEquals(id100, mappedIdsPairs.get(2).getTo()),
+                () -> assertEquals(id99, mappedIdsPairs.get(1).getFrom()),
+                () -> assertEquals(id99, mappedIdsPairs.get(1).getTo()),
+                () -> assertEquals(id98, mappedIdsPairs.get(0).getFrom()),
+                () -> assertEquals(id98, mappedIdsPairs.get(0).getTo()));
+    }
+
+    @Test
+    void queryingSameIdMultipleWillReturn1Result() throws SolrServerException, IOException {
+        var id50 = "UPI0000000050";
+
+        var mappedIdsPairs =
+                idMappingRepository.getAllMappingIds(
+                        SolrCollection.uniparc, List.of(id50, id50, id50));
+
+        assertAll(
+                () -> assertEquals(1, mappedIdsPairs.size()),
+                () -> assertEquals(id50, mappedIdsPairs.get(0).getFrom()),
+                () -> assertEquals(id50, mappedIdsPairs.get(0).getTo()));
+    }
+}
```

Comment:
could you please add integration test when few ids are found in solr and few are not.
I noticed that you have already added this scenario with mock in SolrJobTaskTest.